skorch/tests/callbacks/test_training.py

```diff
@@ -487,6 +493,89 @@ def test_typical_use_case_nonstop(
 
         assert len(net.history) == max_epochs
 
+    def test_weights_restore(
+            self, net_clf_cls, classifier_module, classifier_data,
+            early_stopping_cls):
+        patience = 3
+        max_epochs = 20
+        seed = 1
+
+        side_effect = []
+
+        def sink(x):
+            side_effect.append(x)
+
+        early_stopping_cb = early_stopping_cls(
+            patience=patience,
+            sink=sink,
+            load_best=True,
+            monitor="valid_acc",
+            lower_is_better=False,
+        )
+
+        # Split dataset to have a fixed validation
+        X_tr, X_val, y_tr, y_val = train_test_split(
+            *classifier_data, random_state=seed)
+        tr_dataset = TensorDataset(
+            torch.as_tensor(X_tr).float(), torch.as_tensor(y_tr))
+        val_dataset = TensorDataset(
+            torch.as_tensor(X_val).float(), torch.as_tensor(y_val))
+
+        # Fix the network once with early stoppping and fixed seed
+        net = net_clf_cls(
+            classifier_module,
+            callbacks=[early_stopping_cb],
+            max_epochs=max_epochs,
+            train_split=predefined_split(val_dataset),
+        )
+        torch.manual_seed(seed)
+        net.fit(tr_dataset, y=None)
+
+        # Check training was stopped before the end
+        assert len(net.history) < max_epochs
+
+        # check correct output messages
+        assert len(side_effect) == 2
+
+        msg = side_effect[0]
+        expected_msg = ("Stopping since valid_acc has not improved in "
+                        "the last 3 epochs.")
+        assert msg == expected_msg
+
+        msg = side_effect[1]
+        expected_msg = ("Restoring best model from epoch ")
```

Comment:
```suggestion
        expected_msg = "Restoring best model from epoch "
```

Do we not know the actual epoch, to include it into the message?

Reply:
We do (14) but I preferred not to include it in case something changes in the `conftest.py` or else at some point and this number changes. The person creating the PR will then be confused about the reason this test will not be passing anymore.